mainmatter/method_of_comparison/api_methods/index.tex

```diff
@@ -0,0 +1,184 @@
+In the case of programming languages, a lot of design choices have no empirical grounding or are backed up by more dubious metrics such as \gls{LOC} \cite{markstrum2010staking}.
+Yet, in recent years more empirical studies of languages have cropped up.
+These are often performed as split tests in controlled settings, where each group of participants are tasked with completing a set number of tasks in a specific language.
+Such an approach was used in \cite{hanenberg2010experiment}, where a comparison of statically and dynamically typed languages was made.
+More substantial metrics than \gls{LOC} were collected from the experiment, such as the average time spent in different phases of development.
+There has also been attempts at analysing large sets of data pulled from online code repositories.
+The authors of \cite{ray2014large} compared 17 languages by pulling code from 728 popular github repositories.
+Each language was profiled and compared by the amount of errors they resulted in, as well as the type of error.
+
+While the amount of literature dealing with programming languages is beginning to become substantial, equivalent literature in the area of \gls{API} design evaluation, has not reached the same point.
+In this section, we cover the three categories of evaluation commonly found in the literature.
+These being user tests, non-user evaluations and program metrics.     
+
+The goal of looking deeper into these different types of evaluation is to put together a mixed-method approach for evaluating Direct3D 12 and Vulkan.
+Such an approach was used in \cite{grill2012methods}\cite{beaton2008usability} to great effect, as the combined usage of methods was able to spot a larger amount of usability problems.
+
+\subsection{User Tests}
+A common approach to evaluate the programmability of an \gls{API} is to observe how actual users wield the tool.
+Thus, the group of methods presented here is heavily inspired by the field of \gls{HCI}.
+
+Traditional user tests involve designers creating a list of tasks, which participating users should accomplish.
+Evaluators facilitate test sessions, where participants are asked to perform the tasks using think-aloud protocol.
+As a session progresses the actions and comments of users are recorded and noted down by evaluators.
+Following a session, participants are often interviewed or answer a questionnaire to post-usage data.
+The collected data is then analyzed to find and rate usability problems.
+
+An issue with this approach is that it can be rather expensive, depending on the number of participants and the type of tasks.
+In addition, the standard process is usually geared towards evaluating \glspl{GUI} and not tools like programming languages.
+Evaluating a programming language differs in that tasks can have wildly different solutions.
+There should also be accounted for learning required to understand a language.
```

Comment:
Der er også en lærings process involveret i at lære at bruge programmer med GUI, plejer man at tage højde for det?